plaidml/op/lib/ops.cc

```diff
@@ -2914,6 +2915,53 @@ Value transpose(const Value& value) {
   return Value{O};
 }
 
+Value unsqueeze(const Value& value) {
+  // Read arguments
+  auto args = value.as_tuple();
+  if (args.size() != 2) {
+    throw std::runtime_error(llvm::formatv("PlaidML unsqueeze op expects 2 arguments (received {0})", args.size()));
+  }
+
+  auto I = args[0].as_tensor();
+  auto ndims = I.rank();
+
+  std::vector<int64_t> raw_axes;
+  if (args[1].is_int()) {
+    raw_axes.push_back(args[1].as_int());
+  } else {
+    raw_axes = args[1].as_int_tuple();
+  }
+  std::set<size_t> axes;
+  for (auto& raw_axis : raw_axes) {
+    axes.insert(normalize_axis(raw_axis, ndims + 1, "unsqueeze"));
+  }
+
+  std::vector<TensorDim> I_dims(ndims);
+  std::vector<TensorIndex> I_idxs;
+  std::vector<TensorDim> O_dims;
+  std::vector<TensorIndex> O_idxs;
+  I.bind_dims(I_dims);
+  size_t src_loc = 0;
+  size_t new_rank = ndims + axes.size();
+  for (size_t i = 0; i < new_rank; i++) {
+    if (axes.count(i)) {
+      O_dims.push_back(edsl::TensorDim(1));
+      O_idxs.emplace_back("a");
```

Comment:
This will give every unsqueeze index the same name. IIRC they are semantically meaningless, so I think this runs correctly, but it will be confusing to anyone using the name to figure out what's happening. Using `llvm::formatv("a{0}", i)` analogously to the `else` clause is probably the simplest way to clean this.